packages/voting/src/reply.rs

```diff
@@ -56,3 +57,33 @@ pub mod error {
         UnknownReplyId { id: u64 },
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_tagged_reply_id() {
+        // max u62, change this if new reply types added
+        let proposal_id = 4611686018427387903;
+        let proposal_hook_idx = 1234;
+        let vote_hook_idx = 4321;
+
+        let m_proposal_id = mask_proposal_execution_proposal_id(proposal_id);
+        let m_proposal_hook_idx = mask_proposal_hook_index(proposal_hook_idx);
+        let m_vote_hook_idx = mask_vote_hook_index(vote_hook_idx);
+
+        assert_eq!(
+            TaggedReplyId::new(m_proposal_id).unwrap(),
+            TaggedReplyId::FailedProposalExecution(proposal_id)
+        );
+        assert_eq!(
+            TaggedReplyId::new(m_proposal_hook_idx).unwrap(),
+            TaggedReplyId::FailedProposalHook(proposal_hook_idx)
+        );
+        assert_eq!(
+            TaggedReplyId::new(m_vote_hook_idx).unwrap(),
+            TaggedReplyId::FailedVoteHook(vote_hook_idx)
+        );
```

Comment:
Love having little tests like this.

Reply:
Same, simple test, that shows how to use this feature.